connectors/sources/notion.py

```diff
@@ -0,0 +1,457 @@
+        attachment_metadata["extension"] = "." + response.url.path.split(".")[-1]
+        attachment_metadata["size"] = response.content_length
+        attachment_metadata["name"] = unquote(response.url.path.split("/")[-1])
+        return attachment_metadata
+
+    async def get_content(self, attachment, file_url, timestamp=None, doit=False):
+        """Extracts the content for Apache TIKA supported file types.
+
+        Args:
+            attachment (dictionary): Formatted attachment document.
+            timestamp (timestamp, optional): Timestamp of attachment last modified. Defaults to None.
+            doit (boolean, optional): Boolean value for whether to get content or not. Defaults to False.
+
+        Returns:
+            dictionary: Content document with _id, _timestamp and attachment content
+        """
+        attachment = await self.get_file_metadata(attachment, file_url)
+        if attachment == []:
+            self._logger.info("skipping attachment")
+        attachment_size = int(attachment["size"])
+
+        attachment_name = attachment["name"]
+        attachment_extension = attachment["extension"]
+
+        if not self.can_file_be_downloaded(
+            attachment_extension, attachment_name, attachment_size
+        ):
+            return
+        document = {
+            "_id": f"{attachment['_id']}",
+            "_timestamp": attachment["_timestamp"],
+        }
+
+        return await self.download_and_extract_file(
+            document,
+            attachment_name,
+            attachment_extension,
+            partial(
+                self.generic_chunked_download_func,
+                partial(self.notion_client.get_via_session, url=file_url),
+            ),
+        )
+
+    def _format_doc(self, data):
+        """Format document for handling empty values & type casting.
+
+        Args:
+            data (dict): Fetched record from Notion.
+
+        Returns:
+            dict: Formatted document.
+        """
+        data = {key: value for key, value in data.items() if value}
+        data["_id"] = data["id"]
+        if "last_edited_time" in data:
+            data["_timestamp"] = data["last_edited_time"]
+
+        if "properties" in data:
+            data["details"] = str(data["properties"])
+            del data["properties"]
+        return data
+
+    def generate_query(self):
+        if self.pages == ["*"] and self.databases == ["*"]:
+            yield {}
+        else:
+            for page in self.pages:
+                yield {
+                    "query": "" if page == "*" else page,
+                    "filter": {"value": "page", "property": "object"},
+                }
+            for database in self.databases:
+                yield {
+                    "query": "" if database == "*" else database,
+                    "filter": {"value": "database", "property": "object"},
+                }
+
+    async def get_docs(self, filtering=None):
```

Comment:
I'd like to see INFO logging either in `get_docs` or one level down to provide indication of how the sync is progressing. Like:
```
[INFO] - fetching Users
...
[INFO] - fetching Pages
...
[INFO] - fetching Databases
```
etc